Custom Bar Glyphs

zsunberg opened the pull request:
Implements idea in #35 taking suggestions in #36 into account

zsunberg commented:
oops - guess I could have just added commits to #36; didn't realize you could do that. Also, let me know if this looks good, and I'll update the README as well

tlycken commented on src/ProgressMeter.jl:
```diff
@@ -252,12 +281,20 @@ function printover(io::IO, s::AbstractString, color::Symbol = :color_normal)
 end
 
-function barstring(barlen, percentage_complete; solidglyph="█", emptyglyph=" ")
+function barstring(barlen, percentage_complete; barglyphs::BarGlyphs=BarGlyphs('|','█','█',' ','|'))
```
Minor style nitpick: here you use the `BarGlyphs(chars...)` constructor, while above (line 78) you use the `BarGlyphs(string)` constructor. Is there a reason not to use the same one?

tlycken commented on test/test.jl:
```diff
@@ -229,6 +230,20 @@ end
+function testfunc14(barglyphs)
+    n = 30
+    # full keyword argumetns
+    p = ProgressMeter.Progress(n, barglyphs=ProgressMeter.BarGlyphs(barglyphs))
```
This is nice. It would be even nicer to add a test that uses this in conjunction with the other kwargs (I don't think it will make a difference now, but we should make sure that it doesn't in the future).

Also, a couple of tests that invoke both constructors of `BarGlyphs` would be nice, since they're part of the public interface.

tlycken commented:
Looks good overall to me! I added a couple of line comments with some minor things to address. Also, would you mind squashing the commits? (If you don't know how to, that's fine anyway.)

Thanks a lot for a great addition!

zsunberg commented:
Ok, addressed those comments and added a demo to the readme. To tired to learn about rebasing (I guess that's what squashing entails) tonight, but will read about it at some point.

tlycken commented:
LGTM, thanks!
